source/structured_data.cc

```diff
@@ -604,6 +607,174 @@ namespace aspect
+                    if (dimids_to_use[i]!=dimids[i])
+                      {
+                        use=false;
+                        break;
+                      }
+                }
+              else
+                {
+                  for (int i=0; i<dim; ++i)
+                    {
+                      dimids_to_use[i] = dimids[i];
+                      size_t length;
+                      status = nc_inq_dim(ncid, dimids[i], nullptr, &length);
+                      new_table_points[i] = length;
+                    }
+                }
+
+              if (use)
+                {
+                  varids_to_use.push_back(varid);
+                  data_column_names.push_back(name);
+                  std::cout << "\tusing variable." << std::endl;
+                }
+              else
+                std::cout << "\tignoring variable with different dims." << std::endl;
+
+            }
+          else
+            std::cout << "\tignoring variable with the wrong number of dims." << std::endl;
+        }
+
+
+      {
+        // Now load coordinate data
+
+        for (int d=0; d<dim; ++d)
+          {
+            int varid = dimids_to_use[d];
+
+            nc_type xtype;
+            int ndims;
+            int dimids[NC_MAX_VAR_DIMS];
+            char  name[NC_MAX_NAME];
+            int natts;
+
+            status = nc_inq_var (ncid, varid, name, &xtype, &ndims, dimids,
+                                 &natts);
+            AssertThrow(ndims == 1, ExcMessage("A variable of a dimension should have only one dimension."));
+
+            AssertThrow(xtype == NC_DOUBLE || xtype == NC_FLOAT, ExcMessage("We only support float or double data."));
+
+            coordinate_values[d].resize(new_table_points[d]);
+            status = nc_get_var_double(ncid, varid, coordinate_values[d].data());
+            AssertThrow(status == NC_NOERR, ExcMessage("NetCDF Error occured!"));
+
+            std::cout << "coordinate " << d << " length= " << new_table_points[d] << " data=  " << coordinate_values[d][0] << ' ' << coordinate_values[d][1] << " ..." << std::endl;
+          }
+      }
+
+      {
+        // Finally load the data for each column
+        data_tables.resize(varids_to_use.size());
+        std::vector<double> raw_data;
+
+        for (unsigned int var = 0; var<varids_to_use.size(); ++var)
+          {
+            // Allocate space
+            data_tables[var].TableBase<dim,double>::reinit(new_table_points);
+            const std::size_t n_elements = data_tables[var].n_elements();
+            raw_data.resize(n_elements);
+
+            // Load the data
+            status = nc_get_var_double(ncid, varids_to_use[var], raw_data.data());
+            AssertThrow(status == NC_NOERR, ExcMessage("NetCDF Error occured!"));
+
+            // .. and copy it over:
+            for (std::size_t n = 0; n < n_elements; ++n)
+              data_tables[var](compute_table_indices(new_table_points, n)) = raw_data[n];
```

Comment:
I guess this is the place you mentioned where you have to check that coordinates run in the same order as for our `Table<*,dim>`?